src/FSharp.Data.GraphQL.Server/Execution.fs

```diff
@@ -507,86 +494,109 @@ let private executeQueryOrMutation (resultSet: (string * ExecutionInfo) []) (ctx
                     k::kvps, e@errs) ([],[])
             return NameValueLookup(dicts |> List.rev |> List.toArray), (errors |> List.rev)
         }
-
+    let rec traversePath (d : DeferredExecutionInfo) (fieldCtx : ResolveFieldContext) (path: string list) (tree: AsyncVal<ResolverTree>) (pathAcc: string list): AsyncVal<(ResolverTree * string list) []> =
+        asyncVal {
+            let! tree' = tree
+            let! res = 
+                match List.tail path, tree' with
+                | [], t -> 
+                    asyncVal { 
+                        let! res = buildResolverTree d.Info.ReturnDef fieldCtx fieldExecuteMap t.Value
+                        match d.Info.Kind with
+                        | SelectFields [f] -> return! async { return [|res, List.rev (f.Identifier :: pathAcc)|] }
+                        | _ -> return! async { return [|res, List.rev ((List.head path)::pathAcc)|] }
+                    }
+                | [p], t -> 
```

Comment:
I think this case and the case below can be combined. i.e. ```| ( [p;"__index"] | [p]), t -> ...```

Reply:
Yes, it's possible. I will be commiting this change.